src/components/join-flow/join-flow.jsx

```diff
@@ -61,59 +64,111 @@ class JoinFlow extends React.Component {
             this.handleSubmitRegistration(this.state.formData);
         });
     }
+    getBodyErrors (err, body, res) {
+        return (!err && res.statusCode === 200 && body && body[0] && body[0].errors);
+    }
+    getSingleError (bodyErrors) {
+        if (Object.keys(bodyErrors).length === 1) {
+            const fieldName = Object.keys(bodyErrors)[0];
+            if (bodyErrors[fieldName].length === 1) {
+                return {fieldName: fieldName, errorStr: bodyErrors[fieldName][0]};
+            }
+        }
+        return null;
+    }
+    getCustomErrMsg (bodyErrors) {
+        let customErrMsg = '';
+        // body can include zero or more error objects, each
+        // with its own key and description. Here we assemble
+        // all of them into a single string, customErrMsg.
+        const errorKeys = Object.keys(bodyErrors);
+        errorKeys.forEach(key => {
+            const vals = bodyErrors[key];
+            vals.forEach(val => {
+                if (customErrMsg.length) customErrMsg += '; ';
+                customErrMsg += `${key}: ${val}`;
+            });
+        });
+        if (!customErrMsg) return null; // if no key-val pairs
+        const problemsStr = this.props.intl.formatMessage({id: 'registration.problemsAre'});
+        return `${problemsStr} "${customErrMsg}"`;
+    }
+    getRegistrationSuccess (err, body, res) {
```

Comment:
Rename something that indicates this returns a bool. is/wasRegistrationSuccessfull, wasUserCreated, or something like that.

Reply:
Done; called it `registrationIsSuccessful`